components/results_table/commons/results_table.lua

```diff
@@ -101,8 +109,203 @@ function BaseResultsTable:getQueryType()
 	error('Invalid querytype "' .. (args.querytype or '') .. '"')
 end
 
+function BaseResultsTable:create()
+	local data = self.args.data or self:queryData()
+
+	if Table.isEmpty(data) then
+		return
+	end
+
+	table.sort(data, function(placement1, placement2) return placement1.date > placement2.date end)
+
+	if self.config.onlyAchievements then
+		self.data = {data}
+		return self
+	end
+
+	-- split data into years for non achievements
+	local splitData = {}
+	for _, placementData in ipairs(data) do
+		local year = placementData.date:sub(1,4)
+		if not splitData[year] then
+			splitData[year] = {header = year}
+		end
+
+		table.insert(splitData[year], placementData)
+	end
+
+	self.data = {}
+	for _, dataSet in Table.iter.spairs(splitData, function(tbl, key1, key2) return key1 > key2 end) do
+		table.insert(self.data, dataSet)
+	end
```

Comment:
reminder for myself: break line

Reply:
- `extractValues` is not needed as it already is an array due to the above comment/change
- schreds the data and produces errors

Reply:
```suggestion
	Array.sortInPlaceBy(splitData, function(val) return val.header end, function(val1, val2) return val1 > val2 end)
	self.data = splitData
```
since:
1. Array.sortInPlaceBy has no return value
2. Array.extractValues is not needed due to it already being an array
3. the 2nd var passed to `Array.sortInPlaceBy` has to be the function that extracts the value needed for the compare, the 3rd one is the compare function

Reply:
You can do `self.data = Array.sortBy(...)` if you want instead